### PR TITLE
Reduce computational time of AWS megatest

### LINKED ISSUE #137: Fix `-resume` with fai usage
### Description of the bug

Using the `fai` element in the `ch_fasta` channel hinder the resume of the pipeline.
When used the cached element aren't used and the pipeline recompute the processes and the succeeding ones.

Possibilities:
- Find why
- Remove the fai as it is not directly used (not even sure it really is used in the different processes)

Example: in the `test` profile the pipeline will not resume the variant calling at the preprocessing of glimpse 1

### Command used and terminal output

nextflow run nf-core/phaseimpute -profile singularity,test --outdir results -resume

### Relevant files

_No response_

### System information

_No response_

### PR DESCRIPTION
**Pull request by LouisLeNezet**
- This PR change the features tested in the fulltest config on AWS to reduce computational cost.
- The snapshot of `VCF_PHASE_SHAPEIT5` is updated as it was not the case. (my bad in the previous PR).

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/phaseimpute/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/phaseimpute _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core pipelines lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by LouisLeNezet** [comment]
Whenever we have merged the chunk_model update I will launch the updated fulltest on my cluster.

**Review by atrigila — CHANGES_REQUESTED** [review]
Always impressed by the amount of work you put on these contributions - much appreciated! :)

I noticed several distinct topics being addressed here. To make reviews faster and keep the changelog and file history cleaner, it would help if each of these topics were split into separate PRs. This also makes it easier to track changes and ensure each update is well-documented. Here are the main topics I see in this PR and that I tried to provide feedback on:

1. Reducing computational cost in fulltest configuration on AWS: This is the primary focus, and it's a great enhancement. You said previously that you have evidence that these new changes make the pipeline full test work, so I believe this is ready. 
2. `VCF_SPLIT_BCFTOOLS` edge case fix: Since this is a specific edge case fix, a separate PR would be ideal. In this particular case I see a great benefit of having this particular subworkflow in nf-core/modules, as it allows the community to review and provide feedback on these changes, speeding up the review process and then only making this a simple update in the pipeline :)
3. Snapshot file updates and output file prefix changes: I understand that there have been changes in how files are named and therefore the snapshots, but it would be nice if this could have been something separate to provide context on why these prefixes were adjusted.
4. "Lenient mode" addition: This appears to be a new feature but I couldn't find any reference in your PR description talking about this, could you explain its purpose and the scenarios where it would be useful?
5. Minor corrections in the changelog and punctuation fixes: These small adjustments (like punctuation or formatting) would be best suited for a separate "cleanup" PR  but they are OK.

File: conf/modules.config [hunk 1]
```diff
@@ -31,11 +31,13 @@ process {
     // Simulation
     // Coverage process
     withName: 'NFCORE_PHASEIMPUTE:PHASEIMPUTE:SAMTOOLS_COVERAGE_INP' {
-        ext.prefix = { "${meta.id}.A-truth" }
+        cache = "lenient"
```
[the remark this answers is not in the document]
Comment (LouisLeNezet) [reply conf/modules.config:34]: The `lenient` mode enable to cache the file without checking for the timestamp of the last modification of the files.
This enable us to resume even with the fasta index that was recomputed some times.
More infos in https://www.nextflow.io/docs/latest/cache-and-resume.html

**Comment by LouisLeNezet** [comment]
I'm sorry that this PR have too much files changed.
It's just that when I do some debugging there is some small changed that needs to be adressed and they pile up quite quickly.
I'll try to split it.

File: nextflow.config
[the remark this answers is not in the document]
Comment (LouisLeNezet) [reply nextflow.config:1]: This change allow to add change in the `test<all,glimpse2,quilt,...>.config` files that would be otherwise overwritten by the default if this was include afterwards.

File: conf/test.config [hunk 2]
```diff
@@ -12,8 +12,8 @@
 
 process {
     resourceLimits = [
-        cpus: 4,
-        memory: '15.GB',
+        cpus: 2,
+        memory: '2.GB',
```
[the remark this answers is not in the document]
Comment (LouisLeNezet) [reply conf/test.config:16]: Sorry I thought it was 2.
I will update it to 4 in this case.

File: assets/multiqc_config.yml [hunk 3]
```diff
@@ -17,3 +17,10 @@ top_modules:
 export_plots: true
 
 disable_version_detection: true
+
+fn_clean_trim:
+  - ".panel"
+
+extra_fn_clean_exts:
+  - type: regex
+    pattern: "\\.batch[0-9]+"
```
[the remark this answers is not in the document]
Comment (LouisLeNezet) [reply assets/multiqc_config.yml:26]: There is batch file processed in the statistic. The aim to remove this pattern was to clean up a bit the name of the sample in the multiqc output html.
What could be done would be to allow the grouping of the samples by batch in the multiqc output (enable by the last multiqc version). But I think this should be done outside this PR.

File: workflows/phaseimpute/tests/main.nf.test.snap [hunk 4]
```diff
@@ -299,28 +299,28 @@
...
-                "simulation/NA19401_depth_1x.bam",
-                "simulation/NA19401_depth_1x.bam.csi",
-                "simulation/NA20359_depth_1x.bam",
-                "simulation/NA20359_depth_1x.bam.csi",
+                "simulation/NA12878.depth_1x.bam",
+                "simulation/NA12878.depth_1x.bam.csi",
+                "simulation/NA19401.depth_1x.bam",
+                "simulation/NA19401.depth_1x.bam.csi",
+                "simulation/NA20359.depth_1x.bam",
+                "simulation/NA20359.depth_1x.bam.csi",
```
[the remark this answers is not in the document]
Comment (LouisLeNezet) [reply workflows/phaseimpute/tests/main.nf.test.snap:315]: This will be done in a separate PR

File: workflows/phaseimpute/tests/main.nf.test.snap [hunk 5]
```diff
@@ -166,63 +166,63 @@
...
                 "validation/samples/NA20359.truth.vcf.gz",
                 "validation/samples/NA20359.truth.vcf.gz.tbi",
+                "validation/stats/AllSamples.txt",
                 "validation/stats/NA12878.truth.bcftools_stats.txt",
+                "validation/stats/NA12878_SNP.txt",
                 "validation/stats/NA19401.truth.bcftools_stats.txt",
-                "validation/stats/NA20359.truth.bcftools_stats.txt"
+                "validation/stats/NA19401_SNP.txt",
+                "validation/stats/NA20359.truth.bcftools_stats.txt",
+                "validation/stats/NA20359_SNP.txt"
```
[the remark this answers is not in the document]
Comment (LouisLeNezet) [reply workflows/phaseimpute/tests/main.nf.test.snap:189]: This will be done in a separate PR

**Review by LouisLeNezet — COMMENTED** [review]
Ready for changes

File: conf/steps/validation.config [hunk 6]
```diff
@@ -65,24 +67,24 @@ process {
...
-        ext.args   = ["--output-type z", "--write-index=tbi"].join(' ')
-        ext.prefix = { "${meta.id}" }
-        ext.suffix = ".truth"
-    }
-
     // Validation subworkflow
     withName: 'NFCORE_PHASEIMPUTE:PHASEIMPUTE:VCF_CONCORDANCE_GLIMPSE2:.*' {
         publishDir = [
-            path: { "${params.outdir}/validation/" },
+            path: { "${params.outdir}/validation/stats" },
```
[the remark this answers is not in the document]
Comment (LouisLeNezet) [reply conf/steps/validation.config:85]: Done in separate PR.

File: conf/steps/panel_prep.config [hunk 7]
```diff
@@ -185,7 +185,7 @@ process {
     }
 
     withName: 'NFCORE_PHASEIMPUTE:PHASEIMPUTE:VCF_CHUNK_GLIMPSE:GLIMPSE_CHUNK' {
-        ext.args   = "--window-size 4"
+        ext.args   = "--window-size 4000000"
```
[the remark this answers is not in the document]
Comment (LouisLeNezet) [reply conf/steps/panel_prep.config:188]: The value of this parameter is empirical and might need to be optimized depending on the cluster and specie studied.
This value seems to be a good compromised : enough parrelization and not to small to crash GLIMPSE2_CHUNKS.

[the remark this answers is not in the document]
Comment (LouisLeNezet) [reply conf/steps/panel_prep.config:188]: Done

**Comment by LouisLeNezet** [comment]
Hi @atrigila this PR is now ready !

File: conf/test_all.config [hunk 8]
```diff
@@ -12,8 +12,8 @@
 
 process {
     resourceLimits = [
-        cpus: 2,
-        memory: '10.GB',
+        cpus: 4,
+        memory: '4.GB',
```
Comment (atrigila) [inline conf/test_all.config:16]: I think it can handle up to 15GB

**Review by atrigila — APPROVED** [review]
